### ISSUE TITLE
How do I get Precision and Recall to be calculated together when I run model_main_tf2.py ?

### ISSUE THREAD

**Issue by tmori2918** [tensorflow/models#9695]
Hello, I usually create my models by using Tensorflow2 Object Detection API and my original data.

When I create a model, I always only see the Loss graph in Tensorboard.
I know that if I run model_main_tf2.py with the option "--checkpoint_dir" after creating the model, Precison and Recall will be calculated.
However, I want to check the progress of learning while calculating Precision and Recall during running model_main_tf2.py.

I remember when I used Tensorflow-gpu1.15 before, the training was going on while Precision and Recall were being calculated.

If you know how to train and create a model while calculating Precision and Recall when running model_main_tf2.py, please let me know.

Thanks!

・OS : Ubuntu20.04LTS
・NVIDIA Driver : 450.102
・CUDA : 11.0
・cuDNN : 8.0.2
・Python : 3.6
・Tensorflow-gpu : 2.4.1

**Comment by jartantupjar** [comment]
You can run it in parallel. Run one without the checkpoint_dir flag and one with it. This will run the evaluation in parallel to your training

**Comment by jartantupjar** [comment]
You may run into some memory issues if you only have 1 GPU. So you may need to configure the model_main_tf2.py file to divide your current GPU into 2 separate GPUs using tf.config.experimental.set_virtual_device_configuration. I used to solve this very problem on my gtx1080ti. You can find my model_main_tf2 file below in txt format
[model_main_tf2.txt](https://github.com/tensorflow/models/files/6023219/model_main_tf2.txt)
